src/audio/up_down_mixer/up_down_mixer.c

```diff
@@ -320,232 +320,136 @@ static int init_mix(struct comp_dev *dev,
-
-	if (dev->state == COMP_STATE_ACTIVE) {
-		comp_info(dev, "up_down_mixer_prepare(): Component is in active state.");
-		return 0;
-	}
-
-	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
-	if (ret < 0)
-		return ret;
-
-	if (ret == COMP_STATUS_STATE_ALREADY_SET)
-		return PPL_STATUS_PATH_STOP;
-
 	return 0;
 }
 
-static int up_down_mixer_reset(struct comp_dev *dev)
+static int up_down_mixer_reset(struct processing_module *mod)
 {
-	comp_set_state(dev, COMP_TRIGGER_RESET);
 	return 0;
 }
 
-static int up_down_mixer_copy(struct comp_dev *dev)
+static int
+up_down_mixer_process(struct processing_module *mod,
+		      struct input_stream_buffer *input_buffers, int num_input_buffers,
+		      struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
-	struct comp_buffer *source, *sink;
-	struct comp_buffer __sparse_cache *source_c, *sink_c;
+	struct up_down_mixer_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 	uint32_t source_bytes, sink_bytes;
 	uint32_t mix_frames;
 
-	struct up_down_mixer_data *cd = comp_get_drvdata(dev);
-
-	comp_dbg(dev, "up_down_mixer_copy()");
-
-	source = list_first_item(&dev->bsource_list, struct comp_buffer,
-				 sink_list);
-	sink = list_first_item(&dev->bsink_list, struct comp_buffer,
-			       source_list);
-
-	source_c = buffer_acquire(source);
-	sink_c = buffer_acquire(sink);
+	comp_dbg(dev, "up_down_mixer_process()");
 
-	mix_frames = MIN(audio_stream_get_avail_frames(&source_c->stream),
-			 audio_stream_get_free_frames(&sink_c->stream));
+	mix_frames = mod->input_buffers[0].size;
 
-	source_bytes = mix_frames * audio_stream_frame_bytes(&source_c->stream);
-	sink_bytes = mix_frames * audio_stream_frame_bytes(&sink_c->stream);
+	source_bytes = mix_frames * audio_stream_frame_bytes(mod->input_buffers[0].data);
+	sink_bytes = mix_frames * audio_stream_frame_bytes(mod->output_buffers[0].data);
 
 	if (source_bytes) {
-		buffer_stream_invalidate(source_c, source_bytes);
-		audio_stream_copy_to_linear(&source_c->stream, 0, cd->buf_in, 0,
+		uint32_t sink_sample_bytes;
+
+		audio_stream_copy_to_linear(mod->input_buffers[0].data, 0, cd->buf_in, 0,
 					    source_bytes /
-					    audio_stream_sample_bytes(&source_c->stream));
+					    audio_stream_sample_bytes(mod->input_buffers[0].data));
 
 		cd->mix_routine(cd, (uint8_t *)cd->buf_in, source_bytes, (uint8_t *)cd->buf_out);
 
-		audio_stream_copy_from_linear(cd->buf_out, 0, &sink_c->stream, 0,
-					      sink_bytes /
-					      audio_stream_sample_bytes(&sink_c->stream));
-		buffer_stream_writeback(sink_c, sink_bytes);
-
-		comp_update_buffer_produce(sink_c, sink_bytes);
-		comp_update_buffer_consume(source_c, source_bytes);
+		sink_sample_bytes = audio_stream_sample_bytes(mod->output_buffers[0].data);
+		audio_stream_copy_from_linear(cd->buf_out, 0, mod->output_buffers[0].data, 0,
+					      sink_bytes / sink_sample_bytes);
+		mod->output_buffers[0].size = sink_bytes;
+		mod->input_buffers[0].consumed = sink_bytes;
```

Comment:
This is intentional?

comp_update_buffer_produce(sink_c, **sink_bytes**);
comp_update_buffer_consume(source_c, **source_bytes**);

->

mod->output_buffers[0].size = **sink_bytes**;
mod->input_buffers[0].consumed = **sink_bytes**;

Reply:
no not at all. Thanks for catching this!